File: scripts/resize-icon.py
from PIL import Image
import os

def resize_icon_to_fill(input_path, output_path, target_size=512, fill_percentage=0.75):
    """
    Resize icon to fill more space while maintaining transparency
    
    Args:
        input_path: Path to input PNG
        output_path: Path to save resized PNG
        target_size: Final canvas size (default 512x512)
        fill_percentage: How much of canvas to fill (0.75 = 75%)
    """
    print(f"📂 Opening: {input_path}")
    
    # Open the image
    img = Image.open(input_path).convert("RGBA")
    
    # Get the bounding box of non-transparent pixels
    bbox = img.getbbox()
    
    if bbox:
        # Crop to content
        img_cropped = img.crop(bbox)
        print(f"✂️  Cropped from {img.size} to {img_cropped.size}")
        
        # Calculate new size (fill 75% of canvas)
        target_content_size = int(target_size * fill_percentage)
        
        # Maintain aspect ratio
        width, height = img_cropped.size
        if width > height:
            new_width = target_content_size
            new_height = int((height / width) * target_content_size)
        else:
            new_height = target_content_size
            new_width = int((width / height) * target_content_size)
        
        # Resize the content
        img_resized = img_cropped.resize((new_width, new_height), Image.Resampling.LANCZOS)
        print(f"🔍 Resized to: {img_resized.size}")
        
        # Create new canvas
        canvas = Image.new('RGBA', (target_size, target_size), (255, 255, 255, 0))
        
        # Center the resized image
        x_offset = (target_size - new_width) // 2
        y_offset = (target_size - new_height) // 2
        
        canvas.paste(img_resized, (x_offset, y_offset), img_resized)
        
        # Save
        canvas.save(output_path, 'PNG')
        print(f"✅ Saved enlarged icon to: {output_path}")
        print(f"📏 Final size: {canvas.size}, Content fills ~{int(fill_percentage*100)}% of space")
        
    else:
        print("❌ No content found in image!")

if __name__ == "__main__":
    input_file = r"C:\Users\ABDOUL JABBAR\Desktop\Nouveau dossier\logi\resources\icon-removebg-preview.png"
    output_file = r"C:\Users\ABDOUL JABBAR\Desktop\Nouveau dossier\logi\resources\icon.png"
    
    # Resize to fill 75% of the 512x512 canvas
    resize_icon_to_fill(input_file, output_file, target_size=512, fill_percentage=0.75)
    
    print("\n🔄 Next steps:")
    print("1. node scripts\\generate-ico.js")
    print("2. npx electron-builder --dir --win")

File: scripts/resize-icon.js
const fs = require('fs');
const path = require('path');
const { createCanvas, loadImage } = require('canvas');

async function resizeIcon() {
    // Try both source paths
    let inputPath = path.join(__dirname, '..', 'resources', 'icon-removebg-preview.png');
    if (!fs.existsSync(inputPath)) {
        inputPath = path.join(__dirname, '..', 'resources', 'icon.png');
    }
    const outputPath = path.join(__dirname, '..', 'resources', 'icon.png');

    console.log('📂 Loading:', inputPath);
    const image = await loadImage(inputPath);

    // 1. Get Bounding Box (Trim transparency)
    // We need to find the actual content to ignore the huge empty margins
    const tempCanvas = createCanvas(image.width, image.height);
    const tempCtx = tempCanvas.getContext('2d');
    tempCtx.drawImage(image, 0, 0);
    const pixels = tempCtx.getImageData(0, 0, image.width, image.height).data;

    let minX = image.width, minY = image.height, maxX = 0, maxY = 0;
    let found = false;

    for (let y = 0; y < image.height; y++) {
        for (let x = 0; x < image.width; x++) {
            const alpha = pixels[(y * image.width + x) * 4 + 3];
            if (alpha > 10) { // Threshold to ignore near-transparent noise
                if (x < minX) minX = x;
                if (x > maxX) maxX = x;
                if (y < minY) minY = y;
                if (y > maxY) maxY = y;
                found = true;
            }
        }
    }

    if (!found) {
        console.error("❌ No content found (blank image)");
        return;
    }

    const contentWidth = maxX - minX + 1;
    const contentHeight = maxY - minY + 1;

    console.log(`✂️ Content Bounds: ${contentWidth}x${contentHeight} at (${minX},${minY})`);

    // 2. Resize to 110% of the canvas
    const targetSize = 512;
    const fillPercentage = 1.10; // 110% as requested! Super big.
    const targetContentSize = targetSize * fillPercentage;

    const scale = Math.min(
        targetContentSize / contentWidth,
        targetContentSize / contentHeight
    );

    const newWidth = Math.floor(contentWidth * scale);
    const newHeight = Math.floor(contentHeight * scale);

    console.log(`🚀 Scaling to: ${newWidth}x${newHeight} (${fillPercentage * 100}%)`);

    // 3. Draw Centered
    const canvas = createCanvas(targetSize, targetSize);
    const ctx = canvas.getContext('2d');

    // Center it
    const x = (targetSize - newWidth) / 2;
    const y = (targetSize - newHeight) / 2;

    // Draw
    ctx.imageSmoothingEnabled = true;
    ctx.imageSmoothingQuality = 'high';

    // Draw only the trimmed content, scaled up
    ctx.drawImage(image, minX, minY, contentWidth, contentHeight, x, y, newWidth, newHeight);

    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(outputPath, buffer);
    console.log(`✅ Saved SUPER-SIZED (Trimmmed + 110%) icon to: ${outputPath}`);
}

resizeIcon().catch(err => {
    console.error('❌ Error:', err);
    process.exit(1);
});
